src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -435,12 +431,19 @@ public function get($id, $invalidBehavior = ContainerInterface::EXCEPTION_ON_INV
 
         try {
             $definition = $this->getDefinition($id);
+            if ($definition instanceof ServiceAwareDefinition) {
+                return $definition->getService();
+            }
         } catch (ServiceNotFoundException $e) {
-            if (ContainerInterface::EXCEPTION_ON_INVALID_REFERENCE !== $invalidBehavior) {
-                return;
+            if (ContainerInterface::EXCEPTION_ON_INVALID_REFERENCE === $invalidBehavior) {
+                throw $e;
             }
 
-            throw $e;
+            return;
```

Comment:
Concerning BC: both `throw` and `return` now happens before the deprecation error...